public/sketch.js

```diff
@@ -124,35 +118,31 @@ function draw() {
     let user = users[id];
     let username = user.username;
 
-    fill(user.color.r, user.color.g, user.color.b);
+    fill(user.color);
 
     // if you haven't ended
-    if (!user.endTime) {
-
-      // keep growing fontsize
-      let growthRate = 32;
-      let duration = (Date.now() - user.startTime) / 1000;
-      let size = duration * growthRate + 16;
-      textSize(size);
-
-      // turn keycode to actual letter
-      // print the letter
-      text(user.keycode, user.xpos, user.ypos);
-    } else {
-      textSize(32);
-      text(user.keycode, user.xpos, user.ypos);
+    for (let keycode=0; keycode < user.pressed.length; keycode++) {
+      let keyinfo = user.pressed[keycode];
+      if (keyinfo) {
+        let size = 32;
+        if (!keyinfo.endTime) {
+          let growthRate = 32;
+          let duration = (Date.now() - keyinfo.startTime) / 1000;
+          size = duration * growthRate + 16;
+        }
+        textSize(size);
+        let keychar = String.fromCharCode(keycode);
+        text(keychar, 500, 500);
```

Comment:
I think we lost some of the random position work that @nicosanin and I added. I can add it back, just want to call it out as we learn to work in a team / communicate in code!